packages/flutter_tools/lib/src/base/context.dart

```diff
@@ -7,6 +7,9 @@ import 'dart:collection';
 
 import 'package:meta/meta.dart';
 
+// TODO(ianh): We should remove AppContext's mechanism and replace it with
+// passing dependencies directly in constructors, methods, etc.
+
```

Comment:
Was an issue for this created? I just noticed this todo and it seems like it'd be a huge change to the tool's codebase.

(I'm also curious to learn why you're not happy anymore with the AppContext mechanism)

Reply:
https://github.com/flutter/flutter/issues/47161

Reply:
thank you!